### PR TITLE
Dispose Disposable or AsyncDisposable plugin on mainwindow close

### PR DESCRIPTION
**Pull request by taooceros**
Dispose Disposable or AsyncDisposable plugin on mainwindow close

### CODE REVIEW

**Comment by jjw24** [comment]
What's the impact this issue is causing currently?

**Comment by taooceros** [comment]
We currently don't have a strategy to let plugins know Flow is closing. This will allow plugins to do something when Flow is exiting.

**Comment by jjw24** [comment]
Ok cool, but otherwise no issues casuee right?

**Comment by taooceros** [comment]
> Ok cool, but otherwise no issues caused right?

The ClipBoard plugin? It is causing Flow hang while exiting

**Comment by jjw24** [comment]
I remmeber seeing some plugins inherit the IDisposable interface, do we currently have no implementation of it?

**Comment by taooceros** [comment]
> I remmeber seeing some plugins inherit the IDisposable interface, do we currently have no implementation of it?

I don't think so. Probably what you remember is ISavable?

**Comment by jjw24** [comment]
Hmm ok , I'm afk but that's fine.

**Comment by jjw24** [comment]
I saw your issue in https://github.com/Xenolphthalein/Flow.Launcher.Plugin.ClipboardHistory/issues/4, maybe update this issue to let @Xenolphthalein know how to use it to fix the problem.

**Comment by taooceros** [comment]
By the way, I think we may need to update the plugin development tutorial after the release of 1.8.0. Quite a lot new features are added, and some old features are not documented (like IReloadable, IResultUpdated, etc.)

**Comment by jjw24** [comment]
> By the way, I think we really need to update the plugin development tutorial after the release of 1.8.0. Quite a lot new features are added, and some old features are not documented (like IReloadable, IResultUpdated, etc.)

Oh yeah, for sure

**Comment by jjw24** [comment]
Going to be a lot of writing 😂

File: Flow.Launcher/MainWindow.xaml.cs [hunk 1]
```diff
@@ -46,10 +47,13 @@ public MainWindow()
             InitializeComponent();
         }
 
-        private void OnClosing(object sender, CancelEventArgs e)
+        private async void OnClosing(object sender, CancelEventArgs e)
```
Comment (jjw24) [inline Flow.Launcher/MainWindow.xaml.cs:50]: What happens when user use sys command exit, will it route the call here?

Comment (taooceros) [reply Flow.Launcher/MainWindow.xaml.cs:50]: sys exit will call MainWindow.Close()

**Review by jjw24 — APPROVED** [review]

**Pull request merged**